deploy/lib/createFunctions.js

```diff
@@ -27,21 +29,62 @@ module.exports = {
       });
   },
 
-  createSingleFunction(func) {
+  validateRuntime(func, existingRuntimes) {
+    const existingRuntimesGroupedByLanguage = existingRuntimes
+      .reduce((r, a) => {
+        r[a.language] = r[a.language] || [];
+        r[a.language].push(a);
+        return r;
+      }, Object.create(null));
+
+    const existingRuntimesByName = Object.values(existingRuntimesGroupedByLanguage)
+      .flat()
+      .reduce((map, r) => {
+        map[r.name] = { status: r.status, statusMessage: r.status_message };
+        return map;
+      }, {});
+
+    const currentRuntime = func.runtime || this.runtime;
+
+    if (Object.keys(existingRuntimesByName).includes(currentRuntime)) {
+      const runtime = existingRuntimesByName[currentRuntime];
+      if (runtime.status !== RUNTIME_STATUS_AVAILABLE) {
+        let warnMessage = `WARNING: Runtime ${currentRuntime} is in status ${runtime.status}`;
+        if (runtime.statusMessage !== null && runtime.statusMessage !== undefined && runtime.statusMessage !== '') {
+          warnMessage += `: ${runtime.statusMessage}`;
+        }
+        console.warn(warnMessage);
```

Comment:
why you don't use [this.serverless.cli.log](https://www.serverless.com/framework/docs/guides/plugins/cli-output) ?